runtime/cudaq/operators/spin_op.cpp

```diff
@@ -159,6 +159,15 @@ spin_handler::to_sparse_matrix(const std::string &pauli_word,
   return matrix;
 }
 
+cudaq::detail::EigenSparseMatrix
```

Comment:
The overload that takes an additional dimensions directory only exists for consistency with other operator classes (bosons) for which this argument is needed so that we don't need to template specialize.